core/src/test/java/ai/timefold/solver/core/impl/score/stream/common/bi/AbstractBiConstraintStreamPrecomputeTest.java

```diff
@@ -0,0 +1,395 @@
+package ai.timefold.solver.core.impl.score.stream.common.bi;
+
+import java.util.List;
+import java.util.function.Function;
+
+import ai.timefold.solver.core.api.score.buildin.simple.SimpleScore;
+import ai.timefold.solver.core.api.score.stream.ConstraintCollectors;
+import ai.timefold.solver.core.api.score.stream.Joiners;
+import ai.timefold.solver.core.api.score.stream.PrecomputeFactory;
+import ai.timefold.solver.core.api.score.stream.bi.BiConstraintStream;
+import ai.timefold.solver.core.impl.score.stream.common.AbstractConstraintStreamTest;
+import ai.timefold.solver.core.impl.score.stream.common.ConstraintStreamImplSupport;
+import ai.timefold.solver.core.impl.score.stream.common.ConstraintStreamPrecomputeTest;
+import ai.timefold.solver.core.impl.util.Pair;
+import ai.timefold.solver.core.testdomain.score.lavish.TestdataLavishEntity;
+import ai.timefold.solver.core.testdomain.score.lavish.TestdataLavishEntityGroup;
+import ai.timefold.solver.core.testdomain.score.lavish.TestdataLavishSolution;
+import ai.timefold.solver.core.testdomain.score.lavish.TestdataLavishValue;
+import ai.timefold.solver.core.testdomain.score.lavish.TestdataLavishValueGroup;
+
+import org.junit.jupiter.api.TestTemplate;
+import org.mockito.Mockito;
+
+public abstract class AbstractBiConstraintStreamPrecomputeTest extends AbstractConstraintStreamTest
+        implements ConstraintStreamPrecomputeTest {
+    protected AbstractBiConstraintStreamPrecomputeTest(ConstraintStreamImplSupport implSupport) {
+        super(implSupport);
+    }
+
+    @Override
+    @TestTemplate
+    public void precompute_filter_0_changed() {
```

Comment:
For the sake of brevity, I'd remove the `precompute_` prefix; the class name already fulfills that role.